core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -147,6 +152,22 @@ private ManifestFile copyManifest(ManifestFile manifest) {
     return newManifests;
   }
 
+  @Override
+  protected Map<Integer, String> updatePartitionStats() {
+    Map<Integer, String> newPartitionStatsLocations = new HashMap<>();
+    int currentSpecID = ops.current().spec().specId();
+    List<PartitionStatsEntry> oldStatsEntries = getOldPartitionStatsBySpec(currentSpecID);
+    PartitionsTable.PartitionMap partitionMap = partitionStatsMap.getPartitionMap(currentSpecID);
+    if (Objects.isNull(partitionMap)) {
+      partitionMap = partitionStatsMap.getEmptyPartitionMap(currentSpecID);
```

Comment:
The transaction might be a schema change transaction (like adding a new column). Does this require creating a new partition stats for the current specif?

Reply:
schema changes will be part of UpdateSchema.java interface. Above code is only for FastAppend operation. We don't require to create new stats file. This PR implementation will not create new partition stats file. as UpdateSchema only updates table metadata.